### ISSUE TITLE
Github CI bot runs tasks for non-contributor PRs as well

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#18975]
My understanding is that the Github CI is meant to run lint/detekt/build for contributor PRs that are not part of the mozilla-mobile organization that have access to run Taskcluster CI builds.

I have PRs where the Github CI _and_ Taskcluster CI are both running in parallel a similar subset of tasks.

Ref issue: https://github.com/mozilla-mobile/fenix/pull/18974
<img width="849" alt="Screen Shot 2021-04-12 at 7 38 05 PM" src="https://user-images.githubusercontent.com/1370580/114475726-c6702400-9bc6-11eb-964e-ec0488c922f5.png">

cc: @st3fan

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-10357)

**Comment by boek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813640

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
